dotnet/src/SemanticKernel.Abstractions/Functions/SKJsonSchema.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Diagnostics;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+#pragma warning disable IDE0130
+// ReSharper disable once CheckNamespace - Using the main namespace
+namespace Microsoft.SemanticKernel;
+
+/// <summary>Represents JSON Schema for describing types used in <see cref="ISKFunction"/>s.</summary>
+[JsonConverter(typeof(SKJsonSchema.JsonConverter))]
+public sealed class SKJsonSchema
+{
+    /// <summary>The schema stored as a string.</summary>
+    private string? _schemaAsString;
+
+    /// <summary>Parses a JSON Schema for a parameter type.</summary>
+    /// <param name="jsonSchema">The JSON Schema as a string.</param>
+    /// <returns>A parsed <see cref="SKJsonSchema"/>.</returns>
+    /// <exception cref="ArgumentException"><paramref name="jsonSchema"/> is null.</exception>
+    /// <exception cref="JsonException">The JSON is invalid.</exception>
+    public static SKJsonSchema Parse(string jsonSchema) => new(JsonSerializer.Deserialize<JsonElement>(jsonSchema));
+
+    /// <summary>Parses a JSON Schema for a parameter type.</summary>
+    /// <param name="jsonSchema">The JSON Schema as a sequence of UTF16 chars.</param>
+    /// <returns>A parsed <see cref="SKJsonSchema"/>.</returns>
+    /// <exception cref="JsonException">The JSON is invalid.</exception>
+    public static SKJsonSchema Parse(ReadOnlySpan<char> jsonSchema) => new(JsonSerializer.Deserialize<JsonElement>(jsonSchema));
+
+    /// <summary>Parses a JSON Schema for a parameter type.</summary>
+    /// <param name="utf8JsonSchema">The JSON Schema as a sequence of UTF8 bytes.</param>
+    /// <returns>A parsed <see cref="SKJsonSchema"/>.</returns>
+    /// <exception cref="JsonException">The JSON is invalid.</exception>
+    public static SKJsonSchema Parse(ReadOnlySpan<byte> utf8JsonSchema) => new(JsonSerializer.Deserialize<JsonElement>(utf8JsonSchema));
```

Comment:
These methods will at the moment accept any valid JSON document (or even just plain values such as `null` or `[]`). Perhaps some validation (or a debug assertion) from JsonSchema.NET might be warranted here?

Reply:
Makes sense.

Reply:
Ah, I can't here.  This core assembly doesn't reference that library.